pkg/systemtemplate/import.go

```diff
@@ -75,7 +75,7 @@ func SystemTemplate(resp io.Writer, agentImage, authImage, namespace, token, url
 		tolerations = templates.ToYAML(taints)
 	}
 
-	if cluster != nil && cluster.Spec.AgentEnvVars != nil {
+	if cluster != nil && cluster.Spec.AgentEnvVars != nil && len(cluster.Spec.AgentEnvVars) > 0 {
```

Comment:
A `nil` array has length 0.
```suggestion
	if cluster != nil && len(cluster.Spec.AgentEnvVars) > 0 {
```